dataclasses_avroschema/utils.py

```diff
@@ -80,6 +80,9 @@ def standardize_custom_type(value: typing.Any) -> typing.Any:
         return tuple(standardize_custom_type(v) for v in value)
     elif issubclass(type(value), enum.Enum):
         return value.value
+    elif callable(getattr(value, "asdict", None)):
+        # object has asdict() method, it's most likely our model
```

Comment:
@marcosschroh's suggestion will work combined with `type` to derive the class from the instance (similar to the handling of `Enum`)
```suggestion
    elif is_pydantic_model(type(value)):
```